plugins/modules/zos_mvs_raw.py

```diff
@@ -3090,6 +3097,49 @@ def get_content(formatted_name, binary=False, from_encoding=None, to_encoding=No
         return stdout
 
 
+def modify_contents(contents):
+    """Return the content of dd_input to a valid form for a JCL program.
+
+    Args:
+        contents (str or list): The string or list with the program.
+
+    Returns:
+        contents: The content in a proper multi line str.
+    """
+    if isinstance(contents, list):
+        contents = "\n".join(contents)
+        contents = add_space(contents)
+    else:
+        if contents[0] != " " or contents[0] != "-":
+            contents = "  {0}".format(contents)
+        contents = list(contents.split("\n"))
+        contents = add_space(contents)
+        contents = "\n".join(contents)
+    return contents
+
+
+def add_space(lines):
+    """Return the array with two spaces at the beggining.
+
+    Args:
+        lines (list): The list with a line of a program.
+
+    Returns:
+        lines: The list in a proper two spaces and the code.
+    """
+    module = AnsibleModule
```

Comment:
I had suspicions on why using AnsibleModule here instead of AnsibleModuleHelper, went a head and tested and looks like failure message is not working as intended.
I tested using the following task: 
```
- name: List data sets matching pattern in catalog,
            save output to a sequential data set. If the data set exists,
            then reuse it, if it does not exist, create it. Returns output as text.
      zos_mvs_raw:
        program_name: idcams
        auth: true
        dds:
          - dd_output:
              dd_name: sysprint
              return_content:
                type: text
                src_encoding: ibm-1047
                response_encoding: iso8859-1
          - dd_input:
              dd_name: sysin
              content: "  DEFINE CLUSTER -
                            (NAME(IMSTESTL.IMS1.RECON4) -
                            CYL(10 10) -
                            FREESPACE(20 20) -
                            INDEXED -
                            KEYS(32 0) -
                            NOERASE -
                            NONSPANNED -
                            NOREUSE -
                            SHAREOPTIONS(3 3) -
                            SPEED -
                            UNORDERED -
                            RECORDSIZE(4086 32600) -
                            VOLUMES(222222) -
                            UNIQUE)"
      register: result

```
and this was the result : 

```
fatal: [zvm]: FAILED! => {"backups": [], "changed": false, "dd_names": [], "msg": "TypeError(\"AnsibleModule.fail_json() missing 1 required positional argument: 'self'\")", "ret_code": {"code": 8}}
```

Reply:
As you can see is throwing a TypeError that is not related to the root cause and ignores your error message